nocmp_test.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package atomic
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNocmpComparability(t *testing.T) {
+	tests := []struct {
+		desc       string
+		give       interface{}
+		comparable bool
+	}{
+		{
+			desc: "nocmp struct",
+			give: nocmp{},
+		},
+		{
+			desc: "struct with nocmp embedded",
+			give: struct{ nocmp }{},
+		},
+		{
+			desc:       "pointer to struct with nocmp embedded",
+			give:       &struct{ nocmp }{},
+			comparable: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			typ := reflect.TypeOf(tt.give)
+			assert.Equalf(t, tt.comparable, typ.Comparable(),
```

Comment:
optional: is it worth adding this as a test to the generated code for the types as well?

Reply:
I've been avoiding generating tests as well. How about we list all generated types in this table just to be sure?